src/Camelotia.Presentation/ViewModels/FolderViewModel.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Camelotia.Presentation.Interfaces;
+using Camelotia.Services.Models;
+using ReactiveUI;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Camelotia.Presentation.ViewModels
+{
+    public class FolderViewModel : ReactiveObject, IFolderViewModel
+    {
+        private readonly FolderModel _folder;
+
+        private readonly IEnumerable<IFolderViewModel> _children;
+
+        public FolderViewModel(IProviderViewModel provider, FolderModel folder)
+        {
+            Provider = provider;
+            _folder = folder;
+            _children = (folder.Children != null && folder.Children.Any())
+                ? folder.Children.Select(f => new FolderViewModel(provider, f))
```

Comment:
Considering that `new FolderViewModel` line. Probably worth introducing a delegate e.g. `CreateFolderViewModel` that would be similar to this one https://github.com/worldbeater/Camelotia/blob/main/src/Camelotia.Presentation/ViewModels/ProviderViewModel.cs#L20 Currently the implementation of the `FolderViewModel` isn't complex and also probably doesn't require mocking in unit tests, but probably worth following the pattern used in the project.

This will require modifications in unit tests and in composition roots of projects targeting platforms other than Avalonia, so ping me if you need help with updating those (e.g. it could be quite complex to modify code targeting UWP in case if you are using a Linux machine I guess)